src/storage/copier/reader_state.cpp

```diff
@@ -249,11 +251,25 @@ void ReaderSharedState::countBlocks() {
 
 std::unique_ptr<ReaderMorsel> ReaderSharedState::getSerialMorsel(DataChunk* dataChunk) {
```

Comment:
Can we remove serial related logic in reader shared state now? as you're setting the pipeline to run in single thread anyways.